### PR TITLE
Add to model.xml several members we wish to expose in the FX

### PR DESCRIPTION
**Pull request by danmosemsft**
Add to model.xml several members we wish to expose in the FX. Done by pulling lines selectively from a model.xml created from the full s.p.corelib.dll. Some of these required new entries in ecalllist.h, I uncommented in one case for this but otherwise commented out the entries in model.xml. I'll submit a follow up PR with these uncommented and the ecalls.

testing: only build.

misc:
	Fix up to date check of BCLRewriter: should build when model.xml is changed
	Add parameter to build.cmd to build both mscorlib and its native image, then stop.

@stephentoub @jkotas @Maoni0 (for exposing GetGenerationWR to Core)

### CODE REVIEW

**Comment by danmosemsft** [comment]
The extra commits listed above may be because I inadvertently submitted my last PR from master... anyone know whether I should do something to correct that..?

**Comment by danmosemsft** [comment]
In adding these, I skipped big areas such as Reflection and areas such as AppDomain we plan to analyze specifically. Also I didn't add the serialization constructor/GetObjectData methods: we intend to add serialization consciously in a follow up change by someone.

**Comment by danmosemsft** [comment]
For future reference, how do I access a Jenkins build output such as D:\j\workspace\debug_windows---17180f1b\bin\Logs\CrossgenCoreLib_Windows_NT__x64__Debug.log?

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build please

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by danmosemsft** [comment]
Aha, the Linux ARM Release failures are a dupe of #6289

File: src/mscorlib/model.xml [hunk 1]
```diff
@@ -514,6 +519,16 @@
...
       <Member Name="TryParse(System.String,System.Char@)" />
     </Type>
+    <Type Name="System.CharEnumerator">
+      <Member Name="Clone" />
+      <Member Name="Dispose" />
+      <Member Name="get_Current" />
+      <Member Name="MoveNext" />
+      <Member Name="Reset" />
+      <Member MemberType="Property" Name="Current" />
+      <Member Status="ImplRoot" Name="System.Collections.IEnumerator.get_Current" />
```
Comment (jkotas) [inline src/mscorlib/model.xml]: These should not be tagged as ImplRoot.

File: src/mscorlib/model.xml [hunk 2]
```diff
@@ -1890,6 +1962,9 @@
       <Member Name="get_Message" />
       <Member Name="get_StackTrace" />
       <Member Name="get_Source" />
+<!--
+      <Member Name="get_TargetSite" />
```
Comment (jkotas) [inline src/mscorlib/model.xml]: Why is this commented out?

File: src/mscorlib/model.xml [hunk 3]
```diff
@@ -6325,11 +6406,19 @@
       <Member Name="#ctor(System.Char[])" />
       <Member Name="#ctor(System.Char[],System.Int32,System.Int32)" />
       <Member Name="#ctor(System.SByte*)" />
+      <Member Name="#ctor(System.SByte*,System.Int32,System.Int32)" />
+<!--
+      <Member Name="#ctor(System.SByte*,System.Int32,System.Int32,System.Text.Encoding)" />
```
Comment (jkotas) [inline src/mscorlib/model.xml]: To enable this one, you need to remove ifdef around `COMString::StringInitSBytPtrPartialEx` in `ecalllist.h`

**Comment by jkotas** [comment]
> exposing GetGenerationWR to Core

We will want to rollback on https://github.com/dotnet/coreclr/pull/5534 once this is exposed (through the contract).

**Comment by jkotas** [comment]
LGTM

**Comment by jkotas** [comment]
> inadvertently submitted my last PR from master... anyone know whether I should do something to correct that

You can keep pushing the PR like this, as long as it gets merged as squash. However, once this get merged you will need to do surgery to get your local master pristine again - something like `git fetch origin` and `git reset --hard origin/master`.

**Comment by stephentoub** [comment]
With the recommended squash, LGTM

**Comment by danmosemsft** [comment]
Should I wait for @Maoni0 on this one, or is it uncontroversial and suffiient to follow up later? (That would make it available to corefx for the interns)

**Comment by danmosemsft** [comment]
Hmm - belay that - I see some CI failures. I'll start building some of these flavors, unless someone can let me know how I'd get to see D:\j\workspace\x86_ryujit_ch---7ecedcb1\bin\Logs\CrossgenCoreLib_Windows_NT__x86__Checked.log

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by jkotas** [comment]
@danmosemsft It is pretty non-contraversial. You can merge as soon as you get green CI.

**Comment by danmosemsft** [comment]
Ryujit checked NT fails with  :     4>EXEC : error : Unable to load the service index for source https://api.nuget.org/v3/index.json. [D:\j\workspace\x86_ryujit_ch---7ecedcb1\src\mscorlib\facade\mscorlib.csproj]

checked Ubuntu: waiting on the others.

arm release Emulator: hudson.plugins.git.GitException: Failed to fetch from https://github.com/dotnet/coreclr.git

arm debug emulator: umount: /opt/linux-arm-emulator-root/dev: device is busy.

@dotnet-bot test  Linux ARM Emulator Cross Debug Build
@dotnet-bot test  Linux ARM Emulator Cross Release Build
@dotnet-bot test Ubuntu x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test

**Comment by danmosemsft** [comment]
arm debug emulator has same problem again. does machine need reboot?

**Comment by danmosemsft** [comment]
Same for the other emulator run. @jkotas  - what is correct procedure to unstick this?

**Comment by jkotas** [comment]
No idea. Looks like nuget snafu, not related to your change.

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by danmosemsft** [comment]
@prajwal-aithal the emulator failures seem to have happened in #6177 and which suggests they were fixed by #6050. Do you have any suggestions?

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/arm_emulator_cross_debug_ubuntu_prtest/559/console
00:01:27.252 + sudo umount /opt/linux-arm-emulator-root/dev
00:01:27.260 umount: /opt/linux-arm-emulator-root/dev: device is busy.
00:01:27.260         (In some cases useful info about processes that use
00:01:27.260          the device is found by lsof(8) or fuser(1))
00:01:27.265 Build step 'Execute shell' marked build as failure

**Comment by danmosemsft** [comment]
@dotnet-bot Windows_NT x64 Release Priority 1 Build and Test

**Comment by Maoni0** [comment]
it's good to expose GetGenerationWR to core but why not other ones (the full GC notification APIs)?

**Comment by jkotas** [comment]
This round is about exposing APIs available in Mono/Xamarin - the full GC notifications are not exposed in Mono/Xamarin.

**Comment by Maoni0** [comment]
I see

**Comment by danmosemsft** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by danmosemsft** [comment]
@jkotas the emulator failures seem to have happened in #6177 and which suggests they were fixed by #6050. I pinged the owner. Would one normally wait while this is resolved, or is this sufficiently unrelated that you would commit? There's no doubt I'm not causing the error, but perhaps if the job ran it would give useful coverage.

**Comment by danmosemsft** [comment]
Yay, they got past that and hit #6289 which is unrelated ("Cannot print exception string because Exception.ToString() failed")

**Comment by prajwal-aithal** [comment]
@danmosemsft That failure is unrelated to your change. We are looking into why this failure is occurring in coreclr and corefx.

**Comment by prajwal-aithal** [comment]
@sjsinju I suggest that we disable the checks temporarily in coreclr and corefx and resolve this error (happening with increased frequency).

**Comment by sjsinju** [comment]
@prajwal-aithal Sure. I would take some action to resolve this error temporarily.

**Pull request merged**